src/framework/ui/checkbox/checkbox.component.tsx

```diff
@@ -46,48 +55,53 @@ export class CheckBox extends React.Component<Props> {
         borderWidth: style.borderWidth,
         borderColor: style.borderColor,
         backgroundColor: style.backgroundColor,
+        ...strictStyles.border,
       },
       icon: {
         width: style.size / 2,
         height: style.size / 8,
         backgroundColor: style.selectColor,
+        ...strictStyles.icon,
       },
       highlight: {
         width: style.highlightSize,
         height: style.highlightSize,
         borderRadius: style.highlightBorderRadius,
         backgroundColor: style.highlightColor,
         opacity: style.highlightOpacity,
+        ...strictStyles.highlight,
       },
     };
   };
 
-  public render(): React.ReactNode {
-    const componentStyle: StyleType = this.getComponentStyle(this.props.themedStyle);
+  public render(): React.ReactElement<TouchableOpacityProps> {
+    const { themedStyle, ...derivedProps } = this.props;
+    const { border, icon, highlight } = this.getComponentStyle(themedStyle);
 
     return (
       <TouchableOpacity
-        {...this.props}
+        {...derivedProps}
         activeOpacity={1.0}
         onPress={this.onPress}
         onPressIn={this.onPressIn}
         onPressOut={this.onPressOut}>
-        <View style={styles.border}>
-          <View style={[styles.highlight, componentStyle.highlight]}/>
-          <View style={[styles.border, componentStyle.border]}>
-            <CheckMark style={componentStyle.icon}/>
+        <View style={strictStyles.border}>
```

Comment:
As long as we don't have `styles` var I do not see a reason why we need rename it.

Reply:
You're mostly right, but I think this is more suitable name for this type of styles
In most cases this is used to prevent override